src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,27 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE_FOR_PRESIDENT_POSITION = 35;
+    private static final String NEEDED_NATIONALITY = "Ukrainian";
+    private static final int MIN_PERIOD = 10;
+    private static final String SEPARATOR = "-";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        int periodCandidateInUkr = getPeriodCandidate(candidate.getPeriodsInUkr());
+        return candidate.getAge() >= MIN_AGE_FOR_PRESIDENT_POSITION
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals(NEEDED_NATIONALITY)
```

Comment:
to avoid NullPointer 
```suggestion
                && NEEDED_NATIONALITY.equals(candidate.getNationality())
```